js/src/builtin/streams/ReadableByteStreamController.cpp

```diff
@@ -761,15 +567,231 @@ enum BYOBRequestSlots {
+  CallArgs args = CallArgsFromVp(argc, vp);
+  Rooted<ReadableByteStreamController*> unwrappedController(
+      cx, UnwrapAndTypeCheckThis<ReadableByteStreamController>(cx, args,
+                                                               "enqueue"));
+  if (!unwrappedController) {
+    return false;
+  }
+
+  // Step 2: If ! ReadableByteStreamControllerCanCloseOrEnqueue(this) is
+  //         false, throw a TypeError exception.
+  if (!js::CheckReadableStreamControllerCanCloseOrEnqueue(
+          cx, unwrappedController, "enqueue")) {
+    return false;
+  }
+
+  // Step 3: Return ! ReadableByteStreamControllerEnqueue(this, chunk).
+  if (!ReadableByteStreamControllerEnqueue(cx, unwrappedController,
+                                           args.get(0))) {
+    return false;
+  }
+  args.rval().setUndefined();
+  return true;
+}
+
+/**
+ * Streams spec, 3.9.4.4. error ( e )
+ */
+static bool ReadableByteStreamController_error(JSContext* cx, unsigned argc,
+                                               Value* vp) {
+  // Step 1: If ! IsReadableByteStreamController(this) is false, throw a
+  //         TypeError exception.
+  CallArgs args = CallArgsFromVp(argc, vp);
+  Rooted<ReadableByteStreamController*> unwrappedController(
+      cx, UnwrapAndTypeCheckThis<ReadableByteStreamController>(cx, args,
+                                                               "enqueue"));
+  if (!unwrappedController) {
+    return false;
+  }
+
+  // Step 2: Perform ! ReadableByteStreamControllerError(this, e).
+  if (!ReadableStreamControllerError(cx, unwrappedController, args.get(0))) {
+    return false;
+  }
+
+  args.rval().setUndefined();
+  return true;
+}
+
+static const JSPropertySpec ReadableByteStreamController_properties[] = {
+    JS_PSG("byobRequest", ReadableByteStreamController_byobRequest, 0),
+    JS_PSG("desiredSize", ReadableByteStreamController_desiredSize, 0),
+    JS_PS_END};
+
+static const JSFunctionSpec ReadableByteStreamController_methods[] = {
+    JS_FN("close", ReadableByteStreamController_close, 0, 0),
+    JS_FN("enqueue", ReadableByteStreamController_enqueue, 1, 0),
+    JS_FN("error", ReadableByteStreamController_error, 1, 0), JS_FS_END};
+
+static const JSClassOps ReadableByteStreamControllerClassOps = {
+    nullptr,                               // addProperty
+    nullptr,                               // delProperty
+    nullptr,                               // enumerate
+    nullptr,                               // newEnumerate
+    nullptr,                               // resolve
+    nullptr,                               // mayResolve
+    ReadableByteStreamControllerFinalize,  // finalize
+    nullptr,                               // call
+    nullptr,                               // construct
+    nullptr,                               // trace
+};
+
+JS_STREAMS_CLASS_SPEC(ReadableByteStreamController, 0, SlotCount,
+                      ClassSpec::DontDefineConstructor,
```

Comment:
This line stops the constructor from being defined however we need to have the constructor defined and exposed as per the WebIDL definition:
```webidl
[Exposed=*]
interface ReadableByteStreamController {
  readonly attribute ReadableStreamBYOBRequest? byobRequest;
  readonly attribute unrestricted double? desiredSize;

  undefined close();
  undefined enqueue(ArrayBufferView chunk
);
  undefined error(optional any e
);
};
```